src/units/unit.cpp

```diff
@@ -1798,9 +1798,17 @@ int unit::resistance_against(const std::string& damage_name,bool attacker,const
 	if(!resistance_abilities.empty()) {
 		unit_abilities::effect resist_effect(resistance_abilities, 100-res);
 
+		bool has_max_value = false;
+		for(auto& i : resistance_abilities) {
+			if((*i.ability_cfg).has_attribute("max_value")){
+				has_max_value = true;
+				break;
+			}
+		}
+		int max_value = has_max_value ? resistance_abilities.highest("max_value").first : 100;
```

Comment:
```
[resistance]
	id=steadfast1
	add=20
	affect_self=yes
	#max_value=90 - default 100
[/resistance]
[resistance]
	id=steadfast2
	add=20
	affect_self=yes
	max_value=50
[/resistance]
[resistance]
	id=steadfast3
	add=20
	affect_self=yes
	max_value=10
[/resistance]
# expected: steadfast3 limits resistances to 10
```
I think currently it limits to 50 instead.

Reply:
Yes,if max_value  defined then default_value  not used, if value of highest attribute who is used for the choisi beteween composite and max_value

Reply:
It should use lowest max_value, and highest min_value.

Reply:
what min_value? resistance can't have min_value in diffrence of [illuminates]